pkg/operator/cluster/ceph/mon/mon.go

```diff
@@ -464,23 +468,41 @@ func (c *Cluster) getAvailableMonNodes() ([]v1.Node, *v1.NodeList, error) {
 	return availableNodes, nodes, nil
 }
 
-func (c *Cluster) getNodesWithMons() (*util.Set, error) {
+func (c *Cluster) getNodesWithMons(nodes *v1.NodeList) (*util.Set, error) {
+	// get the mon pods and their node affinity
 	options := metav1.ListOptions{LabelSelector: fmt.Sprintf("app=%s", appName)}
 	pods, err := c.context.Clientset.CoreV1().Pods(c.Namespace).List(options)
 	if err != nil {
 		return nil, err
 	}
-	nodes := util.NewSet()
+	nodesInUse := util.NewSet()
 	for _, pod := range pods.Items {
 		hostname := pod.Spec.NodeSelector[apis.LabelHostname]
 		logger.Debugf("mon pod on node %s", hostname)
-		nodes.Add(hostname)
+		name, ok := getNodeNameFromHostname(nodes, hostname)
+		if !ok {
+			logger.Errorf("mon %s on hostname %s not found in node list", pod.Name, hostname)
+		}
+		nodesInUse.Add(name)
+	}
+	return nodesInUse, nil
+}
+
+// Look up the immutable node name from the hostname label
+func getNodeNameFromHostname(nodes *v1.NodeList, hostname string) (string, bool) {
+	for _, node := range nodes.Items {
+		if node.Labels[apis.LabelHostname] == hostname {
+			return node.Name, true
+		}
+		if node.Name == hostname {
+			return node.Name, true
```

Comment:
could there be a case where the node name is equal to the hostname selector but different than the node hostname label? Im asking to see whether we should bother checking the node.name and just rely on the node label. Or we can just return the `node.Name` after we have exhausted all the nodes and couldnt find a matching hostname label.

Reply:
i don't see how that case would be possible with the hostname selector being different than the hostname label when the node name is equal. If we have exhausted all possibilities, then we will log an error and continue with other nodes.

Reply:
OK. LGTM